pkg/kubectl/cmd/get.go

```diff
@@ -513,3 +527,50 @@ func RunGet(f cmdutil.Factory, out, errOut io.Writer, cmd *cobra.Command, args [
 	cmdutil.PrintFilterCount(errOut, len(objs), filteredResourceCount, len(allErrs), filterOpts, options.IgnoreNotFound)
 	return utilerrors.NewAggregate(allErrs)
 }
+
+func cmdSpecifiesOutputFmt(cmd *cobra.Command) bool {
+	return cmd.Flags().Lookup("output") != nil && cmdutil.GetFlagString(cmd, "output") != ""
+}
+
+// outputOptsForMappingFromOpenAPI looks for the output format metatadata in the
+// openapi schema and returns the output options for the mapping if found.
+func outputOptsForMappingFromOpenAPI(f cmdutil.Factory, openAPIcacheDir string, mapping *meta.RESTMapping) *printers.OutputOptions {
+
+	// user has not specified any output format, check if OpenAPI has
+	// default specification to print this resource type
+	api, err := f.OpenAPISchema(openAPIcacheDir)
+	if err != nil {
+		// Error getting schema
+		return nil
+	}
+	// Found openapi metadata for this resource
+	kind, found := api.LookupResource(mapping.GroupVersionKind)
+	if !found {
+		// Kind not found, return empty columns
+		return nil
+	}
+
+	columns, found := openapi.GetPrintColumns(kind.Extensions)
+	if !found {
+		// Extension not found, return empty columns
+		return nil
+	}
+
+	return outputOptsFromStr(columns)
+}
+
+// outputOptsFromStr parses the print-column metadata and generates printer.OutputOptions object.
+func outputOptsFromStr(columnStr string) *printers.OutputOptions {
+	if columnStr == "" {
+		return nil
+	}
+	parts := strings.SplitN(columnStr, "=", 2)
+	if len(parts) < 2 {
+		return nil
```

Comment:
This should be returning `(opts *OutputOptions, ok bool)` in accordance with Golang style guides.